cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -456,8 +462,22 @@ private void upgradeBusinessMetadataDatasetSpec() {
     // later if we see this as a more frequent pattern.
     // For now, update the type using Gson, and deserialize the updated JsonObject as a new DatasetSpecification.
     Gson gson = new Gson();
+    // change the dataset type name since in 3.3  it changed to MetadataDataset
     JsonObject jsonObject = gson.toJsonTree(oldBusinessMetadataSpec, DatasetSpecification.class).getAsJsonObject();
```

Comment:
what is the point here? why not deserialize as DatasetSpec, then create a modified datasetSpec, and serialize that for writing back?